Add open fec API instuctions

LindsayYoung opened the pull request:
Thought this might be helpful for @ccostino and others

LindsayYoung committed:
Add open fec API instuctions

Thought this might be helpful for @ccostino and others

jmcarp commented:
Is this preferred way to install our fork of swagger-ui? Wouldn't running `npm install` from the api directory be enough? Also, if we want to encourage a clone and symlink workflow, `npm link` is probably more idiomatic.

jmcarp commented:
By the way, I wonder if we should rename this to `18f/fec-swagger-ui` or something that conveys that this isn't the 18f-wide swagger-ui fork.

cc @noahmanger

noahmanger commented:
:+1:

ccostino commented on README.md:
```diff
@@ -1,3 +1,12 @@
+# FEC API Swagger
+
+This fork creates the interactive documentaion found at: [https://api.open.fec.gov/developers/](https://api.open.fec.gov/developers/)
```
D'oh!  "documentation" is mispelled. :-)

ccostino commented:
Thank you, @LindsayYoung!  @jmcarp, if this module is listed as a dependency in the `package.json` file of the API repo then that sounds like a straight `npm install` would do the trick.  It looks like this is the case in looking over there:  https://github.com/18F/openFEC/blob/develop/package.json#L3

@LindsayYoung, is the idea that by cloning this repo and referencing it directly with a symlink you'd always be able to have the latest updates available via a quick `git pull` as opposed to going through the process of updating the npm module?

@jmcarp, if we have an 18F-wide fork of Swagger UI, it probably does make sense to rename this so folks know it is FEC-specific.  And if that's the case, why do we have an FEC-specific fork instead of using the 18F-wide one (just curious is all)?

jmcarp commented:
@ccostino: we don't have an 18f-wide fork of swagger-ui, to my knowledge--I'm just suggesting that we rename this repo to be clear that this is specifically for the beta.fec.gov docs. This fork includes custom styles and hard-codes FEC-specific language, so it wouldn't be appropriate for use outside FEC. We could also look into a cleaner way to customize swagger-ui per project, but it doesn't seem like a high priority.

ccostino commented:
Gotcha, thanks @jmcarp!  I suspected as much with the custom styles and hard-coded language tailored for the FEC but wanted to verify.  I also realized shortly after asking the question about an 18F-wide version that it might be better to ping the front-end guild about that instead.  Either way, renaming this repo makes sense to me for the reasons stated above.

LindsayYoung commented:
Feel free to add better instructions!

I was writing my workflow, which has worked for me when I want to make changes in this repo and try them out locally. I don't think it is necessary if you just want to work on the API. 

Making the name more obvious sounds like a good idea as well!

ccostino commented:
Awesome, thanks @LindsayYoung!

ccostino committed:
Updated README.md instructions.

I have added several updates based on @LindsayYoung's initial instructions and the discussion that the rest of the team had in response.

ccostino commented:
Instructions have been updated based on our discussion!

/cc @LindsayYoung, @jmcarp, @noahmanger
